File: app/src/main/java/cn/com/szw/lib/myframework/utils/preview/PreviewObject.java
package cn.com.szw.lib.myframework.utils.preview;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by 史忠文
 * on 2017/6/9.
 */

public  class PreviewObject implements AbsImg, Parcelable {

    /**
     * @return   设置缩略图
     */
    @Override
    public String getThumbnailUrl() {
        return "";
    }
    /**
     * @return   设置正常图
     */
    @Override
    public String getNormalUrl() {
        return "";
    }
    /**
     * @return   设置大图
     */
    @Override
    public String getLargeUrl() {
        return "";
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }

    public PreviewObject() {
    }

    protected PreviewObject(Parcel in) {
    }
    public static final Creator<PreviewObject> CREATOR = new Creator<PreviewObject>() {
        @Override
        public PreviewObject createFromParcel(Parcel in) {
            return new PreviewObject(in);
        }

        @Override
        public PreviewObject[] newArray(int size) {
            return new PreviewObject[size];
        }
    };

}

File: Flibrary/src/main/java/com/szw/framelibrary/view/preview/PreviewObject.kt
package com.szw.framelibrary.view.preview

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by 史忠文
 * on 2017/6/9.
 */

open class PreviewObject : AbsImg, Parcelable {

    /**
     * @return   设置缩略图
     */
    override val thumbnailUrl: String
        get() = ""
    /**
     * @return   设置正常图
     */
    override val normalUrl: String
        get() = ""
    /**
     * @return   设置大图
     */
    override val largeUrl: String
        get() = ""

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {}

    constructor() {}

    protected constructor(`in`: Parcel) {}

    companion object {
        val CREATOR: Parcelable.Creator<PreviewObject> = object : Parcelable.Creator<PreviewObject> {
            override fun createFromParcel(`in`: Parcel): PreviewObject {
                return PreviewObject(`in`)
            }

            override fun newArray(size: Int): Array<PreviewObject?> {
                return arrayOfNulls(size)
            }
        }
    }

}
